### DISCUSSION TITLE
Android TV

### DISCUSSION THREAD

**Discussion by smarshnz** [strongswan/strongswan#210] [Q&A]
Hi, is it possible to build this for Android TV for easy compatibility with the NVidia Shield, or MiBox for instance? It is no longer available on the android tv play store and has to be side-loaded. It then of course can not be easily accessed through an icon.. (leanback icon issue if my Googling is correct)

Cheers
Steve

**Comment by tobiasbrunner** [comment]
> It is no longer available on the android tv play store  and has to be side-loaded

I don't think it ever was available as it never satisfied the requirements set by Google. It also uses features (e.g. the Storage Access Framework) that some Android TV images don't provide and its UI is not optimized for use on Android TV. See [this comment](https://wiki.strongswan.org/issues/3626#note-5) in the old issue tracker for some details.

**Comment by Yoshi1981** [comment]
Hey yes better Android TV Google TV 10-12 Version please

But this Work With APK Editor in XML 

category android:name="android.intent.category.LEANBACK_LAUNCHER"

and this in Profile Detail 

android:completionThreshold="1"

and this With Google TV 320x180 PNG 

http://atvlauncher.trekgonewild.de/

This Work With Tasker and Strongswan in XML or ATV Generator

Better when add the Strongswan in Play Store TV please

Thanks

**Comment by fancywriter** [comment]
Hello @tobiasbrunner 

I understand it's probably never gonna be supported by Google Play for TV, but installing it with APK should be possible.
And adding support for icon should be easy and straightforward. And I don't see big risk/costs associated with it.

ChatGPT even offered me a solution https://chatgpt.com/share/f5276e68-3568-4e37-b247-aa5d4d92d40a
It requires few modification of source code and rebuild.

Didn't try this yet though... I need some time to understand how to properly build it first.
